### DISCUSSION TITLE
how to disable treesitter in large file

### DISCUSSION THREAD

**Discussion by athena-1114** [numToStr/Comment.nvim#272] [Q&A] [answered]
Comment will call up treesitter, so how i disable Comment plugin in large file, or disable Comment use treesitter in large file

**Comment by numToStr** [comment, marked as answer]
What's the problem with using treesitter? If you are not using treesitter, then it'll fallback to using `filetype`.

**Reply by athena-1114** [reply]
using treesitter on large files will wait a long time，can set an option to use filetype or treesitter ?

**Reply by numToStr** [reply]
Interesting...How large is this file? Just curious.

---

Also, to avoid treesitter from being used completely, you can use this:

```lua
require('Comment').setup({
    pre_hook = function (ctx)
        return require('Comment.ft').get(vim.bo.filetype, ctx.ctype) or vim.bo.commentstring
    end
})
```

If you want, you can also conditionally avoid treesitter for a particular filetype.

**Reply by HUAHUAI23** [reply]
many thanks, this my another github account, the file size is about 3.2m，it's echart.js  `https://www.jsdelivr.com/package/npm/echarts`

**Reply by numToStr** [reply]
3.2m is really quite big 🤣. If this fixes your issue, then please mark your answer :)

**Reply by HUAHUAI23** [reply]
yeah,it fixes my issue, i will mark this later with my another github account.
